Remove useless code samples and add missing ones

curquiza opened the pull request:
I created [scripts to manage code samples](https://github.com/meilisearch/integration-automations/pull/164) (internal only)

1. I found out the following code samples are still in this repo but not used by the documentation anymore, so I removed them:

```bash
meilisearch-dart
- 'faceted_search_2' not found in documentation
- 'get_keys_1' not found in documentation
- 'documents_guide_add_movie_1' not found in documentation
- 'getting_started_communicating_with_a_protected_instance' not found in documentation
- 'faceted_search_facets_1' not found in documentation
```

2. I found the following code samples were missing:

```bash
meilisearch-dart
- 'get_typo_tolerance_1' not found
- 'update_typo_tolerance_1' not found
- 'reset_typo_tolerance_1' not found
- 'typo_tolerance_guide_1' not found
- 'typo_tolerance_guide_2' not found
- 'typo_tolerance_guide_3' not found
- 'typo_tolerance_guide_4' not found
- 'async_guide_canceled_by_1' not found
- 'facet_search_1' not found
- 'facet_search_3' not found
```
All of them were missing, because the related features are implemented.

curquiza commented:
@ahmednfwela @brunoocasali be careful before fixing this kind of issue: https://github.com/meilisearch/meilisearch-dart/issues/340 -> the code samples were not implemented, so the full issue was not done 😊

ahmednfwela commented:
yes my bad, sometimes I create 3/4 prs a day and forget to add the code samples 😅

curquiza commented:
bors try

ahmednfwela commented:
@curquiza I suggest also fixing the link at the top to 
https://github.com/meilisearch/documentation/blob/main/.code-samples.meilisearch.yaml

ahmednfwela commented:
I made a PR that supersedes this https://github.com/meilisearch/meilisearch-dart/pull/358
@curquiza

curquiza commented:
> @curquiza I suggest also fixing the link at the top to
> https://github.com/meilisearch/documentation/blob/main/.code-samples.meilisearch.yaml

I changed the description. Before a kind of documentation existed at this link. I removed it. Not all the code samples are present in https://github.com/meilisearch/documentation/blob/main/.code-samples.meilisearch.yaml



> I made a PR that supersedes this #358 @curquiza

Thank you!
However, can we still merge this current PR? The goal is to provide an up to date documentation to the users as soon as possible
@brunoocasali will review the other PR when he will come back from Holiday 😊

ahmednfwela commented:
yes sure! we can merge this PR now, but is there a place that contains all the code samples ?

ahmednfwela reviewed (approved):
bors merge

ahmednfwela reviewed (approved):
bors merge
